src/ComponentHistory.jsx

```diff
@@ -199,125 +214,153 @@ if (!blocksChanges) {
+            {blocksChanges
+              .slice(
+                0,
+                state.showAllCommits
+                  ? blocksChanges.length
+                  : COMMIT_DISPLAY_LIMIT
               )
-                State.update({ lineCountDeleted, lineCountInserted });
-            },
-          }}
-        />
+              .map((blockHeight, key) => (
+                <button
+                  type="button"
+                  data-selected={state.blockHeight == blockHeight}
+                  onClick={() => {
+                    State.update({ blockHeight });
+                  }}
+                >
+                  <Text as="span" size="text-s" weight="500" color="sand12">
+                    {key === 0 ? `#${blockHeight} (head)` : `#${blockHeight}`}
+                  </Text>
+                  <Text as="span" size="text-xs">
+                    {getTimestampFromBlockHeight(blockHeight)}
+                  </Text>
+                </button>
+              ))}
+          </Commits>
+
+          {!state.showAllCommits &&
+            blocksChanges.length > COMMIT_DISPLAY_LIMIT && (
+              <Widget
+                src="${REPL_ACCOUNT}/widget/DIG.Button"
+                props={{
+                  fill: "outline",
+                  variant: "secondary",
+                  label: "Show All Commits",
+                  size: "small",
+                  style: { width: "100%" },
+                  onClick: () => {
+                    State.update({ showAllCommits: true });
+                  },
+                }}
+              />
+            )}
+        </Sidebar>
       )}
 
-      {state.selectedTab == "render" && (
-        <div>
+      <Main>
+        {state.selectedTab === "history" && (
+          <CommitHeader>
+            <Text>Changes:</Text>
+            <Badges>
+              <Badge backgroundColor="green4" textColor="green11">
+                +{state.lineCountInserted}
+              </Badge>
+              <Badge backgroundColor="red3" textColor="red11">
+                -{state.lineCountDeleted}
+              </Badge>
+            </Badges>
+
+            <Text size="text-s" style={{ marginRight: "auto" }}>
+              {getTimestampFromBlockHeight(state.blockHeight)}
+            </Text>
+          </CommitHeader>
+        )}
+
+        {state.selectedTab === "code" && (
           <Widget
-            code={currentCode}
-            key={`preview-${state.blockHeight}`}
-            props={props}
+            src={`bozon.near/widget/WidgetHistory.CodeHistory`}
+            key={`code-${state.blockHeight}`}
+            props={{
+              pathToWidget: props.widgetPath,
+              currentBlockHeight: state.blockHeight,
+              prevBlockHeight: state.blockHeight,
+            }}
           />
-        </div>
-      )}
-    </Main>
+        )}
+
+        {state.selectedTab === "history" && (
+          <Widget
+            src={`bozon.near/widget/WidgetHistory.CodeHistory`}
```

Comment:
Is this account is fine for testnet too?

Reply:
Great point - we've never had a `testnet` version of this component and it will continue to cause issues in `testnet`. I opened a separate issue here to track this: https://github.com/near/near-discovery-components/issues/534